samples/runtime/qpic-simulator/host/Program.qs

```diff
@@ -86,11 +84,7 @@ namespace Microsoft.Quantum.Samples {
             using ((a, b) = (Qubit(), Qubit())) {
                 H(a);
                 let result = M(a);
-                if (result == One) {
-                    X(b);
-                } else {
-                    Z(b);
-                }
+                Extensions.ApplyIfElseR(result, (X, b), (Z, b));
```

Comment:
The APIs at https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.simulation.quantumprocessor.extensions are not documented; could you add a small comment here explaining what's happening?